admin/widgets.rb

```diff
@@ -0,0 +1,171 @@
+    column I18n.t('active_admin.widget.standard'), :default, sortable: :default do |widget|
+      widget.default ? "Ja" : "Nein"
+    end
+    column I18n.t('active_admin.widget.css_classes'), :css_name
+    column I18n.t('active_admin.widget.id'), :id_name
+    column I18n.t('active_admin.widget.access') do |widget|
+      Goldencobra::Permission.restricted?(widget) ? raw("<span class='secured'>beschränkt</span>") : ""
+    end
+    column "" do |widget|
+      result = ""
+      result += link_to(t(:edit), edit_admin_widget_path(widget), class: "member_link edit_link edit", title: I18n.t('active_admin.widget.title_widget_edit'))
+      result += link_to(t(:delete), admin_widget_path(widget), method: :DELETE, "data-confirm" => t("delete_article", scope: [:goldencobra, :flash_notice]), class: "member_link delete_link delete", title: I18n.t('active_admin.widget.title_widget_delete'))
+      raw(result)
+    end
+  end
+
+  show title: :title do
+    panel I18n.t('active_admin.widget.widget') do
+      attributes_table_for widget do
+        [:title, :content, :css_name, :active].each do |a|
+          row a
+        end
+      end
+    end
+    panel I18n.t('active_admin.widget.articles') do
+      table do
+        tr do
+          ["Title", "url_name"].each do |ta|
+            th ta
+          end
+        end
+        widget.articles.each do |wa|
+          tr do
+            [wa.title, wa.url_name].each do |watd|
+              td watd
+            end
+          end
+        end
+      end
+    end
+  end
+
+  member_action :revert do
+    @version = PaperTrail::Version.find(params[:id])
+    if @version.reify
+      @version.reify.save!
+    else
+      @version.item.destroy
+    end
+    redirect_to :back, notice: "#{I18n.t('active_admin.widget.revert_notice')} #{@version.event}"
+  end
+
+  batch_action :destroy, false
+
+  action_item :undo, only: :edit do
+    if resource.versions.last
+      link_to(I18n.t('active_admin.widget.undo'), revert_admin_widget_path(id: resource.versions.last), class: "undo")
+    end
+  end
+
+  member_action :duplicate do
+    original_widget = Goldencobra::Widget.find(params[:id])
+    new_widget_id = original_widget.duplicate!
+    if Goldencobra::Widget.find(new_widget_id)
+      redirect_to admin_widget_path(id: new_widget_id),
+                  notice: "#{I18n.t('active_admin.widget.success_duplicating_notice')}"
+    else
+      redirect_to :back, notice: "#{I18n.t('active_admin.widget.error_duplicating_notice')}"
+    end
+  end
+
+  action_item :duplicate, only: :edit do
+    link_to(I18n.t('active_admin.widget.duplicate'),
+            duplicate_admin_widget_path(id: resource), class: "duplicate")
+  end
+
+  controller do
+    def show
+      show! do |format|
+         format.html { redirect_to edit_admin_widget_path(@widget) }
+      end
+    end
+  end
+
+  action_item :prev_item, only: [:edit, :show] do
+    render partial: '/goldencobra/admin/shared/prev_item'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.